esmvaltool/diag_scripts/land_carbon_cycle/extraUtils.py

```diff
@@ -0,0 +1,311 @@
+                 nticks=10,
+                 cbtitle='',
+                 col_scale='linear',
+                 tick_locs=[],
+                 ex_tend='both',
+                 cb_or='horizontal',
+                 spacing='uniform'):
+    '''
+    Plots the colorbar to the axis given by axcol_. Uses arrows on two sides.
+    '''
+
+    axco1 = plt.axes(axcol_)
+    if col_scale == 'linear':
+        cb = mpl.colorbar.ColorbarBase(axco1,
+                                       cmap=cm2,
+                                       norm=mpl.colors.BoundaryNorm(
+                                           bounds__, cm2.N),
+                                       boundaries=bounds__,
+                                       orientation=cb_or,
+                                       drawedges=False,
+                                       extend=ex_tend,
+                                       ticks=bounds__[1:-1],
+                                       spacing=spacing)
+        if len(tick_locs) == 0:
+            tick_locator = mpl.ticker.MaxNLocator(nbins=nticks,
+                                                  min_n_ticks=nticks)
+        else:
+            tick_locator = mpl.ticker.FixedLocator(tick_locs)
+        cb.locator = tick_locator
+        cb.update_ticks()
+    if col_scale == 'log':
+        cb = mpl.colorbar.ColorbarBase(axco1,
+                                       cmap=cm2,
+                                       norm=mpl.colors.BoundaryNorm(
+                                           bounds__, cm2.N),
+                                       boundaries=bounds__,
+                                       orientation=cb_or,
+                                       extend=ex_tend,
+                                       drawedges=False,
+                                       ticks=bounds__[1:-1],
+                                       spacing=spacing)
+        if cb_or == 'horizontal':
+            tick_locs_ori = cb.ax.xaxis.get_ticklocs()
+        else:
+            tick_locs_ori = cb.ax.yaxis.get_ticklocs()
+        tick_locs_bn = []
+        for _tl in tick_locs:
+            tlInd = np.argmin(np.abs(bounds__[1:-1] - _tl))
+            tick_locs_bn = np.append(tick_locs_bn, tick_locs_ori[tlInd])
+        if cb_or == 'horizontal':
+            cb.ax.xaxis.set_ticks(tick_locs_bn)
+            cb.ax.xaxis.set_ticklabels(tick_locs)
+        else:
+            cb.ax.yaxis.set_ticks(tick_locs_bn)
+            cb.ax.yaxis.set_ticklabels(tick_locs)
+    cb.ax.tick_params(labelsize=cbfs, size=2, width=0.3)
+    cb.outline.set_alpha(0.)
+    cb.outline.set_color('white')
+    cb.outline.set_linewidth(0 * cblw)
+    for t in cb.ax.get_yticklabels():
+        t.set_fontsize(cbfs)
+        t.set_rotation(cbrt)
+
+    for t in cb.ax.get_xticklabels():
+        t.set_fontsize(cbfs)
+        t.set_rotation(cbrt)
+        t.set_y(-0.02)
+    if cbtitle != '':
+        cb.ax.set_title(cbtitle, fontsize=1.3 * cbfs)
+    return cb
```

Comment:
nice - you have a `utils` after all - I think a few other quasi-repetitive functions can be placed here (like the common mask one, savefig etc) - have a look, if you can plug them in here and just call them for each diag

Reply:
It's part of the plotUtils now.